ROADMAP.md

```diff
@@ -0,0 +1,185 @@
+
+## Ecosystem
+
+### Remake `compat-table` used in preset-env
+
+> https://github.com/kangax/compat-table could use a remake, ideally work with browser vendors on this
+    
+- There is also https://github.com/mdn/browser-compat-data
+- Also use data from test262?
+- Run tests against real browsers
+- have a data-only format
+- Need continued maintainence
+
+### Polyfill behavior
+
+> This is regarding https://github.com/babel/babel/tree/master/packages/babel-preset-env#usebuiltins-usage
+
+- Allow any substitute polyfill instead of `core-js`. You should be able to override anything (custom `Promise`, etc)
+- Make `"usage"` option the default after it is stable.
+
+### Build/publish workflow
+
+- Guide on compiling/publishing ES2015+, .mjs, etc: https://twitter.com/philwalton/status/908082461799616512
+- Support multi-build/folder outputs based on ES version/browser/engines?
+- Includes making sure you can run Babel safely over node_modules [like with #6280](https://github.com/babel/babel/pull/6280). [create-react-app has an issue for this](https://github.com/facebookincubator/create-react-app/issues/3777)
+
+### Codemods for TC39 Proposals
+
+> Lebab/others are already used to convert from ES5 -> ESNext, so incorporate it into Babel itself.
+
+- Refactor [Lebab](https://github.com/lebab/lebab/issues/138) as Babel transforms (can keep the cli since it's a separate tool)
+  - Usecase: ES3 -> ES6+ (on source code)
+  - Usecase: Remove usage of dropped proposals
+  - Usecase: Auto upgrade to the latest version of a proposal spec (if possible)
+  - Can we somehow combine forces in: babel-codemod/jscodeshift/lebab, prettier/recast/babel-generator? I really don't want to update all of these: new syntax equals re-writing the printer in all of these places separately/out of sync.
+
+## Increasing the quality of community plugins
+
+- Work with the community to create guides on how to write plugins or understand ASTs, etc
+- Analysis of API's/syntax used (Google BigQuery)
+- Have #blessed/sanctioned/curated packages according to some standard
+  - Can use for smoke tests
+  - Official testing package
+  - Certain level of coverage, downloads, etc
+  - Create a scoped namespace on github/npm for these? like webpack-contrib
+  - Can enforce linting rules on apis?
+  - Makes ecosystem changes easier if can notify and upgrade these plugins beforehand
+  - Create a set of standard tests to verify against (handles all syntax)
+  - Documented/tested set of options
+
+### ASTExplorer
+
+- allow custom version of babel-standalone (same as REPL to allow per PR tests)
+- integrate with repl? (both are in react)
+- auto publish a plugin to npm?
+- create tests?
+- hook into codesandbox?
+
+---
+
+## Feature
+
+### Performance table for ES6+ (used in babel-preset-env under new option e.g `perf`)
+
+> Add a new option in `preset-env`: will continue to compile if the native support is *slower*. Can setup a threshold, may need to compare against the size difference.
```

Comment:
How will this interact with `loose` or `spec`?

`loose` is pretty common in the wild: https://github.com/facebook/react/blob/145986ec17cbd136849d8c9ef1157830233e00f0/.babelrc#L12

Reply:
(Granted, that's not using env, but people do use env mode)